### ISSUE TITLE
Linker makes the app crash.

### ISSUE THREAD

**Issue by joduss** [xamarin/xamarin-macios#7689]
### Description.

Our application is working when the _linker behaviour_ is set to "_none_" but some part are crashing if it is set to "Link Framework SDK's only". 

If the app is in debug mode, but linker set to "Link Framework SDK's only", these crashes doesn't occur.

A view controller loads programmatically a view, which loads internally from a nib. The nib contains a view containing subviews arranged by various constraints. These constraints have outlets and I'm setting their constant depending on the content.

The target is ios 12.0

From the device console, this message appears:
```
Unhandled Exception:
Foundation.MonoTouchException: Objective-C exception thrown.  Name: NSInvalidArgumentException Reason: -[UIView setConstant:]: unrecognized selector sent to instance 0x145f94040
```

The manual symbolication of the device's crash report gives us the following:
[Logs.zip](https://github.com/xamarin/xamarin-macios/files/4036538/Logs.zip)

```
[see attached zip forfull log]

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Triggered by Thread:  0

Application Specific Information:
abort() called

Last Exception Backtrace:
0   CoreFoundation                	0x19cfc198c __exceptionPreprocess + 228
1   libobjc.A.dylib               	0x19c19a9f8 objc_exception_throw + 55
2   CoreFoundation                	0x19cede1c8 -[NSObject+ 217544 (NSObject) doesNotRecognizeSelector:] + 139
3   UIKitCore                     	0x1c962c220 -[UIResponder doesNotRecognizeSelector:] + 279
4   CoreFoundation                	0x19cfc71d4 ___forwarding___ + 1407
5   CoreFoundation                	0x19cfc8e6c _CF_forwarding_prep_0 + 91
6   MYAPP       					0x1010850ec wrapper_managed_to_native_ObjCRuntime_Messaging_objc_msgSend_intptr_intptr_System_nfloat + 139
7   MYAPP       					0x1010afb68 Xamarin_iOS_UIKit_NSLayoutConstraint_set_Constant_System_nfloat + 4438888 (NSLayoutConstraint.g.cs:216)
8   MYAPP       					0x102699334 mono_jit_runtime_invoke + 1099
9   MYAPP       					0x1027322f4 mono_runtime_invoke_checked + 147
10  MYAPP       					0x1027357ac mono_runtime_invoke + 155
11  MYAPP       					0x100c786b0 native_to_managed_trampoline_3(objc_object*, objc_selector*, _MonoMethod**, bool, unsigned int) + 18096 (registrar.m:96)
12  MYAPP       					0x100c84394 -[AssessmentVC viewWillAppear:] + 66452 [... see attached zip for full log]
```

### Steps to Reproduce

1. Unknown

### Expected Behavior

Should not crash

### Actual Behavior

Crashes

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.3.11 (build 1)
Installation UUID: b440b68c-843e-4f4f-aad2-2bef8911fe7b
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)

	Package version: 604000208

=== Mono Framework MDK ===

Runtime:
	Mono 6.4.0.208 (2019-06/07c23f2ca43) (64-bit)
	Package version: 604000208

=== NuGet ===

Version: 5.3.0.6192

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.0.101/Sdks
SDK Versions:
	3.0.101
	3.0.100
	2.2.300
	2.2.204
	2.2.104
	2.1.700
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.301
	2.1.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.1
	3.0.0
	2.2.5
	2.2.2
	2.1.14
	2.1.13
	2.1.11
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.1.1
	2.0.5

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin Designer ===

Version: 16.3.0.256
Hash: 8a223bfd7
Branch: remotes/origin/d16-3
Build date: 2019-11-01 21:02:02 UTC

=== Xamarin.Android ===

Version: 10.0.6.2 (Visual Studio Community)
Commit: xamarin-android/d16-3/c407838
Android SDK: /Users/jonathanduss/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.1 (API level 25)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 29.0.4
SDK Build Tools Version: 29.0.0

Build Information: 
Mono: mono/mono/2019-06@476d72b9e32
Java.Interop: xamarin/java.interop/d16-3@5836f58
LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
LibZip: nih-at/libzip/rel-1-5-1@b95cf3fd
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/jonathanduss/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 1.4.0.65
Hash: c33b107
Branch: remotes/origin/d16-3
Build date: 2019-11-19 20:33:22 UTC

=== Android Device Manager ===

Version: 1.2.0.116
Hash: d2b2af0
Branch: remotes/origin/d16-3
Build date: 2019-11-19 20:33:42 UTC

=== Apple Developer Tools ===

Xcode 11.3 (15712)
Build 11C29

=== Xamarin.Mac ===

Version: 6.6.0.12 (Visual Studio Community)
Hash: e3c2b406d
Branch: xcode11.2
Build date: 2019-11-01 00:12:07-0400

=== Xamarin.iOS ===

Version: 13.6.0.12 (Visual Studio Community)
Hash: e3c2b406d
Branch: xcode11.2
Build date: 2019-11-01 00:12:08-0400

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 803110001
Git revision: 6ee6ad2ec46ae5a08a1999ee4c815ac656a35b91
Build date: 2019-12-05 16:09:27+00
Build branch: release-8.3
Xamarin extensions: 56bd70ef2e327f71c615cfc29a47fd50685fadcb

=== Operating System ===

Mac OS X 10.14.6
Darwin 18.7.0 Darwin Kernel Version 18.7.0
    Sun Dec  1 18:59:03 PST 2019
    root:xnu-4903.278.19~1/RELEASE_X86_64 x86_64


```

### Build Logs

[Logs.zip](https://github.com/xamarin/xamarin-macios/files/4036538/Logs.zip)

**Comment by chamons** [comment]
Thank you for your feedback!

For us to investigate this further, could you please provide a test case to reproduce your issue. 

We look forward to hearing from you!

**Comment by joduss** [comment]
I passed quite some more time to investigate what was happening and creating a demo project with a similar configuration to put here. 

**Final word:** 
This was my mistake. I had an NSLayoutConstraint outlet, to which was incorrectly attached a UIView. It certainly happened a few weeks ago when I reworked the view and removed and re-add constraints on the view in the xib.

**This still raise the question, why it is working when using no linker?**
